Competition2017/src/competition/subsystems/drive/commands/FieldOrientedTankDriveWithJoystick.java

```diff
@@ -0,0 +1,69 @@
+package competition.subsystems.drive.commands;
+
+import java.util.function.DoubleSupplier;
+
+import competition.operator_interface.OperatorInterface;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import xbot.common.controls.sensors.XJoystick;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.XPropertyManager;
+
+public class FieldOrientedTankDriveWithJoystick extends BaseDriveCommand {
+
+    private final PoseSubsystem poseSubsystem;
+    
+    private final XJoystick leftJoystick;
+    
+    private final PIDManager headingDrivePid;
+    public final double defaultPValue = 1 / 80d;
+ 
+    private ContiguousHeading targetHeading;
+    protected DoubleSupplier targetHeadingSupplier;
+    
+    public FieldOrientedTankDriveWithJoystick(
+            OperatorInterface oi, 
+            DriveSubsystem driveSubsystem, 
+            PoseSubsystem poseSubsystem,
+            XPropertyManager propManager,
+            RobotAssertionManager assertionManager) {
+        super(driveSubsystem);
+        this.poseSubsystem = poseSubsystem;
+        this.leftJoystick = oi.leftJoystick;
+        headingDrivePid = new PIDManager("Rotate to heading", propManager, assertionManager, defaultPValue, 0, 0);
+        
+        headingDrivePid.setErrorThreshold(3);
```

Comment:
why set these statically in code here, if you provide these defaults in the constructor instead it makes it so there's properties we can change to tune things on the robot which might be more flexible?